owslib/feature/__init__.py

```diff
@@ -206,6 +248,131 @@ def getGETGetFeatureRequest(
 
         return base_url + data
 
+    def getPOSTGetFeatureRequest(
+        self,
+        typename=None,
+        filter=None,
+        bbox=None,
+        featureid=None,
+        featureversion=None,
+        propertyname=None,
+        maxfeatures=None,
+        storedQueryID=None,
+        storedQueryParams=None,
+        outputFormat=None,
+        method="Post",
+        startindex=None,
+        sortby=None,
+    ):
+        """Formulate proper GetFeature request using KVP encoding
+        ----------
+        typename : list
+            List of typenames (string)
+        filter : string
+            XML-encoded OGC filter expression.
+        bbox : tuple
+            (left, bottom, right, top) in the feature type's coordinates == (minx, miny, maxx, maxy)
+        featureid : list
+            List of unique feature ids (string)
+        featureversion : string
+            Default is most recent feature version.
+        propertyname : list
+            List of feature property names. '*' matches all.
+        maxfeatures : int
+            Maximum number of features to be returned.
+        method : string
+            Qualified name of the HTTP DCP method to use.
+        outputFormat: string (optional)
+            Requested response format of the request.
+        startindex: int (optional)
+            Start position to return feature set (paging in combination with maxfeatures)
+        storedQueryID : string
+            A name identifying a prepared set available in WFS-service.
+            WFS version 2.0.0 and above only.
+        storedQueryParams : dict
+            Variable amount of extra information sent to server related to
+            storedQueryID to further define the requested data.
+            WFS version 2.0.0 and above only.
+            {'parameter_name': parameter_value}
+        sortby: list (optional)
+            List of property names whose values should be used to order
+            (upon presentation) the set of feature instances that
+            satify the query.
+
+        There are 5 different modes of use
+
+        1) typename and bbox (simple spatial query)
+        2) typename and filter (==query) (more expressive)
+        3) featureid (direct access to known features)
+        4) storedQueryID and optional storedQueryParams
+        5) filter only via Post method
+        """
+
+        try:
+            base_url = next(
+                (
+                    m.get("url")
+                    for m in self.getOperationByName("GetFeature").methods
+                    if m.get("type").lower() == method.lower()
+                )
+            )
+        except StopIteration:
+            base_url = self.url
+
+        if not typename and filter:
+            return base_url, filter
+
+        request = self.create_post_request()
```

Comment:
I did not understand why using a create_post_request function instead of directly inserting the version-dependent Postrequest object creation. Is this function intended to be used anywhere else ?

Reply:
Well, my aim here was to make the post request creation more testable. It being separated makes it testable by itself.

I chose to make a factory with create_post_request(), so `getPOSTGetFeatureRequest()` can be used for wfs1.1.0 and 2.0.0 without needing another function that does almost the same thing but not quite, which also seems to be what `getGetGetFeatureRequest()` is aiming to become.

So this way, getFeature(), whether it's with 1.1.0 or 2.0.0, calls the same function, which checks the WFS version, and builds the appropriate request format. It will also make it easier (I hope) to integrate the next WFS version by simply extending the `postrequest.py` module so it can format post requests for that new version. All without needing to change `getPOSTGetFeatureRequest()` or creating a new function.

On the plus side, `postrequest.py` can also be used by itself to build a request and use it with resquests.post(), which could be useful in identifying if an error is caused by the request itself or if the problem is somewhere else in OWSLib.